File: src/main/java/spaceEngineers/commands/InteractionArgs.java
package spaceEngineers.commands;

public class InteractionArgs {
    public int interactionType;
    public int page;
    public int slot;
    public boolean allowSizeChange;

    public InteractionArgs(InteractionType interactionType, int slot, int page, boolean allowSizeChange) {
        this.interactionType = interactionType.getValue();
        this.slot = slot;
        this.page = page;
        this.allowSizeChange = allowSizeChange;
    }

    public InteractionArgs(InteractionType interactionType, int slot, int page) {
        this(interactionType, slot, page, false);
    }

    public InteractionArgs(InteractionType interactionType, int slot) {
        this(interactionType, slot, -1);
    }

    public InteractionArgs(InteractionType interactionType) {
        this(interactionType, -1);
    }
}

File: src/commonMain/kotlin/spaceEngineers/commands/InteractionArgs.kt
package spaceEngineers.commands

import spaceEngineers.model.ToolbarLocation
import kotlin.jvm.JvmOverloads

class InteractionArgs @JvmOverloads constructor(
    val interactionType: Int,
    val slot: Int = -1,
    val page: Int = -1,
    val allowSizeChange: Boolean = false,
    val itemName: String? = null
) {
    constructor(
        interactionType: InteractionType,
        slot: Int = -1,
        page: Int = -1,
        allowSizeChange: Boolean = false
    ) : this(interactionType.value, slot, page, allowSizeChange)

    constructor(
        interactionType: InteractionType,
        toolbarLocation: ToolbarLocation,
        allowSizeChange: Boolean = false
    ) : this(interactionType.value, toolbarLocation.slot, toolbarLocation.page, allowSizeChange = allowSizeChange)

    constructor(
        interactionType: InteractionType,
        slot: Int = -1,
        page: Int = -1,
        itemName: String
    ) : this(interactionType.value, slot, page, false, itemName)


    companion object {
        fun equip(toolbarLocation: ToolbarLocation): InteractionArgs {
            return InteractionArgs(InteractionType.EQUIP, toolbarLocation)
        }
    }
}